scripts/tuple_tree_generator/tuple_tree_generator/generators/python.py

```diff
@@ -79,6 +93,125 @@ def _python_type(cls, resolved_type: Definition):
+
+        return real_type
+
+    def type_hint(self, field: StructField) -> str:
+        real_type = self._real_type(field)
+        possible_values = None
+        if real_type in BUILTINS_CLASSES:
+            hint_type = BUILTINS_CLASSES[real_type]
+            ctor = "native"
+        elif real_type in [e.name for e in self.schema.enum_definitions()]:
+            hint_type = "str"
+            ctor = "enum"
+            possible_values = f"{real_type}"
+        elif isinstance(field.resolved_type, ReferenceDefinition):
+            hint_type = "Reference"
+            ctor = "parse"
+        elif real_type in [
+            *(s.name for s in self.schema.struct_definitions() if s.abstract),
+            *(self.string_types),
+        ]:
+            hint_type = real_type
+            ctor = "parse"
+        else:
+            hint_type = real_type
+            ctor = "class"
+
+        assert hint_type
+
+        is_sequence = isinstance(field.resolved_type, SequenceDefinition)
+        if isinstance(field.resolved_type, SequenceDefinition):
+            is_abstract = isinstance(field.resolved_type.element_type, UpcastableDefinition)
+        else:
+            is_abstract = isinstance(field.resolved_type, UpcastableDefinition)
+
+        return (
+            f'{{"type": {hint_type}, '
+            + (f'"possibleValues": {possible_values},' if possible_values is not None else "")
+            + f'"ctor": "{ctor}", '
+            + f'"optional": {self.to_bool(field.optional)}, '
+            + f'"isArray": {self.to_bool(is_sequence)}, '
+            + f'"isAbstract": {self.to_bool(is_abstract)}}}'
+        )
+
+    @staticmethod
+    def to_bool(_input: bool) -> str:
+        return "True" if _input else "False"
+
+    def get_mixins(self, struct_name: str) -> str:
+        for mixins_path in self.mixins_paths:
+            with open(Path(mixins_path), "r") as mixins:
+                mixins_parsed = ast.parse(mixins.read())
+                mixin_classes = [
+                    node.name for node in mixins_parsed.body if isinstance(node, ast.ClassDef)
+                ]
+
+                for mc in mixin_classes:
+                    if struct_name == mc.removesuffix("Mixin"):
+                        return mc
+
+                if "AllMixin" in mixin_classes:
+                    return "AllMixin"
+        raise RuntimeError("Couldn't find matching Mixin class")
+
+    def get_mixins_imports(self) -> List[str]:
+        imports = []
+        for mixins_path in self.mixins_paths:
+            with open(Path(mixins_path), "r") as mixins:
+                mixins_parsed = ast.parse(mixins.read())
+                module_classes = [
+                    node.name for node in mixins_parsed.body if isinstance(node, ast.ClassDef)
+                ]
+                imports.append(
+                    f"from {self._get_mixins_import_path(mixins_path)} import {', '.join(module_classes)}"  # noqa: E501
+                )
+        return imports
```

Comment:
A couple of notes:
1. `get_mixins` will get called a bunch of times
2. `get_mixins` and `get_mixins_imports` overlap a bit

I'd make a function such as `_parse_mixins` which is called in `__init__` which takes care of doing `ast` etc., so that both functions don't need to re-parse the mixin file(s) later